### PR TITLE
catch me if you can.

### PR DESCRIPTION
**Pull request by LMT337**
good luck yo!

### CODE REVIEW

File: myJoin.pl [hunk 1]
```diff
@@ -18,68 +18,67 @@
...
 
 #set input flags
-GetOptions ("H" => \$header,
-            "unique" => \$unique,
-            "duplicate" => \$duplicate,
-            "help" => \$help) or die ("Error in command line arguments -h for help\n");
+GetOptions("H" => \$header,
+"unique" => \$unique,
+"duplicate" => \$duplicate,
+"help" => \$help) or die("Error in command line arguments -h for help\n");
```
Comment (ahwagner) [inline myJoin.pl:24]: Tabs are way cleaner! I think this was a test.

File: myJoin.pl [hunk 2]
```diff
@@ -18,68 +18,67 @@
...
 
 #return help flag
-help_me() if ($help);
+help_me()  ($help);
```
Comment (ahwagner) [inline myJoin.pl]: Definitely a test. I see this!

File: myJoin.pl [hunk 3]
```diff
@@ -18,68 +18,67 @@
...
 
-#check for incorrect usage of -d and -u
+#check for incorrect usage -d and -u
```
Comment (ahwagner) [inline myJoin.pl]: maybe this was intentional. I'll leave it up to you if you want to keep it in a subsequent commit.

File: myJoin.pl [hunk 4]
```diff
@@ -18,68 +18,67 @@
...
 die "Error in command line arguments --h for help\n" if ($duplicate && $unique);
 
 #command line input
 foreach(@ARGV){
-	line_hash1($_);
+    line_hash1($_);
```
Comment (ahwagner) [inline myJoin.pl:34]: Solid fix!

File: myJoin.pl (hunk 3 above)
Comment (LMT337) [inline myJoin.pl]: Typo, I'll take it out.

File: myJoin.pl [hunk 5]
```diff
@@ -18,68 +18,67 @@
...
 }
 
 #add hash keys to uniq and dup arrays
-if ($header){
+if($header){
```
Comment (ahwagner) [inline myJoin.pl]: Diff makes it easy to catch these little spacing changes. for consistency, we should leave the space in.

File: myJoin.pl [hunk 6]
```diff
@@ -18,68 +18,67 @@
...
     print "$header_text\n";
 }
 
 foreach my $line(sort keys %line1){
-	print "$line\n" if ($unique && $line1{$line} == 1);
-	print "$line\n" if ($duplicate && $line1{$line} > 1);
-	print "$line\n" if (!$unique && !$duplicate);
+    print "$line\n" if ($unique && $line1{$line} == 1);
+    print "$line\n" if ($duplicate && $line1{$line} > 1);
+    print "$line\n" if (!$unique  !$duplicate);
```
Comment (ahwagner) [inline myJoin.pl]: Good tab editing.

File: myJoin.pl [hunk 7]
```diff
@@ -18,68 +18,67 @@
...
-
-#Use first observed line as header if flagged, populate hash with line name as hash key.
-#Duplicate line names removed keys cannot be duplicated in hash.
+#use first observed line as header if flagged, populate hash with line name as key
+#Duplicate line names removed, keys can not be duplicated in hash.
 #increment any names repeated more than once, indicates they are duplicates.
+
 sub line_hash1{
-	my $input = shift;
+    my $input = shift;
```
Comment (ahwagner) [inline myJoin.pl:56]: This is good, keep this.

File: myJoin.pl [hunk 8]
```diff
@@ -18,68 +18,67 @@
...
     open(FILE, $input) or die $!;
     if ($header){
-    	$header_text = <FILE>;
-    	chomp $header_text;
+        $header_text = <FILE>;
+        chomp $header_text;
```
Comment (ahwagner) [inline myJoin.pl:60]: Yay Xcode!

File: myJoin.pl [hunk 9]
```diff
@@ -18,68 +18,67 @@
...
     }
-	while (<FILE>){
-		chomp;
-		$line1{$_} += 1;
+    while (<FILE>){
+        chomp;
+        $line1{$_} += 1;
```
Comment (ahwagner) [inline myJoin.pl:64]: More tab awesomeness!

File: myJoin.pl [hunk 10]
```diff
@@ -18,68 +18,67 @@
...
     }
-	close FILE;
+    close FILE;
```
Comment (ahwagner) [inline myJoin.pl:66]: Keep this.

File: myJoin.pl [hunk 11]
```diff
@@ -18,68 +18,67 @@
...
+usage: myjoin [options] file1 file2 [file3...]
+options:
+    -h, --help this help text
+    -H  files contain a header line
+    -u  returns list of unique lines[mutually exclusive with -d]
+    -d  returns list of duplicate lines [mutually exclusive with -u]
+    ";
     exit;
 }
+
```
Comment (ahwagner) [inline myJoin.pl:84]: Good.

File: myJoin.pl (hunk 6 above)
Comment (LMT337) [inline myJoin.pl]: line45:
    print "$line\n" if (!$unique  !$duplicate);
    print "$line\n" if (!$unique &&  !$duplicate);

**Pull request merged**